### PR TITLE
Resolve problems with the logic for evaluating missing changelogs

### PR DESCRIPTION
**Pull request by awharn**
Resolves various issues with the logic that was being used to evaluate whether or not changelog modifications were missing.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -32,32 +34,42 @@ async function checkChangelog() {
...
     const lernaPackages = JSON.parse(await execAndReturnOutput(`npx lerna changed --json --loglevel silent`));
     var errors = "";
-    var changedLocal = false;
-    var headerFoundLocal = false;
     for (const package of lernaPackages) {
-      const fileLocation = path.relative(directory, package.location);
+      const changelogLocation = path.join(path.relative(directory, package.location), file);
+      let changedLocal = false;
+      let headerFoundLocal = false;
+      let found = false;
```
Comment (t1m0thyj) [inline index.js:41]: Is `found` a redundant variable? It's value always seems identical to `changedLocal`.

**Review by t1m0thyj — CHANGES_REQUESTED** [review]

File: index.js (hunk 1 above)
Comment (awharn) [reply index.js:41]: Yep, that seems to be the case.

File: index.js [hunk 2]
```diff
@@ -8,7 +8,9 @@ const eventPath = process.env.GITHUB_EVENT_PATH
 const header = core.getInput('header');
 const file = core.getInput('file');
 const lerna = (core.getInput('lerna').toLowerCase() === "true");
-var changed = false;
+
+
+var changed = undefined;
```
[the remark this answers is not in the document]
Comment (awharn) [reply index.js:13]: Would you suggest doing the same thing as headerFound?

[the remark this answers is not in the document]
Comment (t1m0thyj) [reply index.js:13]: > Would you suggest doing the same thing as headerFound?

That's probably the easiest way to fix it 😛

**Review by t1m0thyj — APPROVED** [review]
LGTM! Thanks for resolving this so quickly 🙂

**Pull request merged**